modules/programs/neovim.nix

```diff
@@ -365,13 +365,14 @@ in {
       in mkMerge (
         # writes runtime
         (map (x: x.runtime) pluginsNormalized) ++ [{
-          "nvim/init.vim" = mkIf (neovimConfig.neovimRcContent != "") {
-            text = neovimConfig.neovimRcContent;
-          };
+          "nvim/init-home-manager.vim" =
+            mkIf (neovimConfig.neovimRcContent != "") {
+              text = neovimConfig.neovimRcContent;
+            };
           "nvim/init.lua" = let
             luaRcContent =
               lib.optionalString (neovimConfig.neovimRcContent != "")
-              "vim.cmd.source ${config.xdg.configHome}/nvim/init.vim"
+              "vim.cmd [[source ${config.xdg.configHome}/nvim/init-home-manager.vim]]"
               + lib.optionalString hasLuaConfig
               config.programs.neovim.generatedConfigs.lua;
```

Comment:
This may break if one or more plugins have `type = "lua"`, since the `vim.cmd [[ ... ]]` string is not `\n`-ended, concatenating anything in `generatedConfigs.lua` directly after `]]`.

Reply:
Maybe it would be better being
```
''
  vim.cmd [[source ${config.xdg.configHome}/nvim/init-home-manager.vim]]
''
```